### PR TITLE
dqlite: 0.2.6 -> 1.0.0

### PR DESCRIPTION
**Pull request by adisbladis**
###### Motivation for this change
I wanted to attempt to use `dqlite`.

Note that the `lxd` build is broken in this PR, we need to await a new release before merging.
I don't feel comfortable backporting API changes from the master branch.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/libraries/sqlite-replication/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  nativeBuildInputs = [ tcl ];
+
+  configureFlags = oldAttrs.configureFlags ++ [
+      "--enable-replication"
+      "--disable-amalgamation"
+      "--disable-tcl"
+  ];
+
+  # TODO: Document how to get these values
```
Comment (adisbladis) [inline pkgs/development/libraries/sqlite-replication/default.nix]: How to properly generate these values?

cc @dtzWill @jokogr

File: pkgs/development/libraries/libco/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "canonical";
+    repo = pname;
+    rev = "b8b70b0cf5d6c6521174001133bb4fde6cce761a";
+    sha256 = "0wf6lhk72iq37h32s8sf373z45cls86rk66dsnq0d130s3clx93q";
+  };
+
+  makeFlags = [
+    "PREFIX=\"$(out)\""
```
Comment (jonringer) [inline pkgs/development/libraries/libco/default.nix]:
```suggestion
    "PREFIX=${placeholder "out"}"
```

File: pkgs/development/libraries/raft/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{ stdenv
+, fetchFromGitHub
+, autoreconfHook
+, pkgconfig
+, libuv
+}:
+
+let
+  pname = "raft";
+  version = "0.9.5";
```
Comment (jonringer) [inline pkgs/development/libraries/raft/default.nix]: i would move the pname and version into the mkDerivation scope, and make it recursive (rec)

Comment (adisbladis) [reply pkgs/development/libraries/raft/default.nix]: I prefer not to use `rec`, it sacrifices overrides for nothing but syntactical convenience.

Comment (jonringer) [reply pkgs/development/libraries/raft/default.nix]: ? you can still use overrideAttrs just fine

File: pkgs/development/libraries/sqlite-replication/default.nix (hunk 1 above)
Comment (jokogr) [reply pkgs/development/libraries/sqlite-replication/default.nix]: These are the commit date and hash, so for `3.29.0+replication3` these would be:
`D 2019-08-19T10:56:12` and `e131dd7676fea1a0e101bfe0f25744bc9c4f9a69` respectively ([source](https://github.com/canonical/sqlite/commit/e131dd7676fea1a0e101bfe0f25744bc9c4f9a69)).

If I remember correctly, I couldn't build it or package it without setting so. Feel free to test if it works without setting those files.

**Comment by jokogr** [comment]
LXD has released a version ([3.17](https://discuss.linuxcontainers.org/t/lxd-3-17-has-been-released/5679)) supporting dqlite.

**Comment by adisbladis** [comment]
@jokogr Thanks for the information. This PR has been updated and should now be merge-ready.

File: pkgs/development/libraries/sqlite-replication/default.nix (hunk 1 above)
Comment (adisbladis) [reply pkgs/development/libraries/sqlite-replication/default.nix]: These are now in the sqlite-replication upstream repo so we don't need to track them in nixpkgs.

**Comment by c0bw3b** [comment]
Obsoleted by #70209

**Pull request closed without merging**